codelabs/getting_started/getting_started.md

```diff
@@ -0,0 +1,422 @@
+
+### Check your work
+
+Now refresh the page. Select the EDIT mode, then tap on one of the buttons. You should see a Blockly editor:
+
+![image](workspace_with_loop.png)
+
+Drag and drop the available loop block to the workspace to test it out.
+
+## Create a custom block
+
+Since this is a music maker app, we want a block that plays sounds. We could create one block per sound, but instead we will create a single block with a dropdown to select which note to play:
+
+![image](play_sound_block.png)
+
+In Blockly, a *block definition* describes how a block looks and behaves. This includes its text, colour, and shape. It may also include which other blocks it can connect to.
+
+Blocks can be defined in either JavaScript or JSON. The developer site has a full article on [how to define a block](https://developers.google.com/blockly/guides/create-custom-blocks/define-blocks).
+
+In this codelab we will simply provide the block definition for you to copy and use.
+
+### Define the sound block
+
+Create a JS file to define a new "play sound" block:
+
+1. Add `sound_blocks.js` file in the `scripts` directory.
+1. Add the following code to `sound_blocks.js`:
+
+```js
+Blockly.defineBlocksWithJsonArray([
+  {
+    "type": "play_sound",
+    "message0": "Play %1",
+    "args0": [
+      {
+        "type": "field_dropdown",
+        "name": "VALUE",
+        "options": [
+          ["C4", "sounds/c4.m4a"],
+          ["D4", "sounds/d4.m4a"],
+          ["E4", "sounds/e4.m4a"],
+          ["F4", "sounds/f4.m4a"],
+          ["G4", "sounds/g4.m4a"],
+          ["A5", "sounds/a5.m4a"],
+          ["B5", "sounds/b5.m4a"],
+          ["C5", "sounds/c5.m4a"]
+        ]
+      }
+    ],
+    "previousStatement": null,
+    "nextStatement": null,
+    "colour": 355
+  }
+]);
+```
+
+Add a script tag to `index.html` to include your new block definition:
+
+```html
+<script src="scripts/sound_blocks.js"></script>
+```
+
+### Add the sound block to the toolbox
+
+Now we can update the toolbox to include the new sound block, by adding `<block type="play_sound"></block>`.
+
+```xml
+<xml id="toolbox" style="display: none">
+  <block type="controls_repeat_ext">
+    <value name="TIMES">
+      <shadow type="math_number">
+        <field name="NUM">5</field>
+      </shadow>
+    </value>
+  </block>
+  <block type="play_sound"></block>
+</xml>
+```
+
+Run the app one more time, and play around with the new "Play (sound)" block. It should look like this:
+
+![image](workspace_with_toolbox.png)
+
+### The block factory
+
+This step discussed how to manually define custom blocks in Blockly. Once you've completed the entire codelab, we recommend that you check out our <a href="https://developers.google.com/blockly/guides/create-custom-blocks/blockly-developer-tools">block factory tool</a>, which helps automate part of this process.
```

Comment:
Any reason for the `<a>` tag here? as opposed to a markdown link? or is that a derivative of the google doc.

Reply:
Just derivative. Fixed.